File: 백준/Silver/1049. 기타줄/기타줄.py
input = open(0).readline

N, M = map(int, input().split())

min_bundle, min_each = 1001, 1001
for i in range(M):
    bundle, each = map(int, input().split())
    
    min_bundle = min(min_bundle, bundle)
    min_each = min(min_each, each)


if min_bundle > min_each * 6:
    print(min_each * N)
else:
    answer = min_bundle * (N // 6)
    answer += min_bundle if min_each * (N % 6) > min_bundle else min_each * (N % 6)
    print(answer)

File: 백준/Silver/1049. 기타줄/기타줄.cc
#include <cstdio>

#define MIN(x, y) ((x) < (y) ? (x) : (y))

int main() { 
  int N, M;
  scanf("%d %d", &N, &M);

  int min_set = 1001, min_each = 1001;
  for (int i = 0; i < M; i++) {
    int set, each;
    scanf(" %d %d", &set, &each);

    min_set = MIN(min_set, set);
    min_each = MIN(min_each, each);
  }

  if (min_set > min_each * 6) {
    printf("%d", min_each * N);
  } else {
    printf("%d", min_set * (N / 6) + MIN(min_set, min_each * (N % 6)));
  }

  return 0;
}